### PR TITLE
provider/consul: API definition

### PR DESCRIPTION
**Pull request by lwander**
This is the core set of features needed to make consul usable as a health/service discovery provider in Spinnaker.

@duftler FYI,
@ttomsu PTAL

### CODE REVIEW

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/model/AgentDefinition.groovy [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+package com.netflix.spinnaker.clouddriver.consul.api.v1.model
+
+class AgentDefinition {
+  Config Config
+  Coord Coord
+  Member Member
+
+  class Config {
```
Comment (ttomsu) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/model/AgentDefinition.groovy]: Lots of stuff in here - alphabetical order seems appropriate. Also, camelCase unless there is a specific need to keep them capitalized.

Comment (lwander) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/model/AgentDefinition.groovy]: Order matches documentation: https://www.consul.io/docs/agent/http/agent.html#agent_self

**Comment by ttomsu** [comment]
LGTM

File: clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/model/AgentDefinition.groovy (hunk 1 above)
Comment (lwander) [inline clouddriver-consul/src/main/groovy/com/netflix/spinnaker/clouddriver/consul/api/v1/model/AgentDefinition.groovy]: BumpyCase due to API constraints

**Pull request merged**